src/serac/physics/tests/dynamic_thermal_adjoint.cpp

```diff
@@ -0,0 +1,225 @@
+{
+  auto thermal = create_heat_transfer(nonlinear_opts, dyn_opts, mat);
+
+  double qoi = 0.0;
+  thermal->outputState();
+  for (int i = 0; i < ts_info.num_timesteps; ++i) {
+    double dt = ts_info.totalTime / ts_info.num_timesteps;
+    thermal->advanceTimestep(dt);
+    thermal->outputState();
+    qoi += computeStepQoi(thermal->temperature(), dt);
+  }
+  int rank;
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+  FiniteElementDual initial_temperature_gradient(thermal->temperature().space(), "init_temp_gradient");
+  initial_temperature_gradient = 0.0;
+  FiniteElementDual shape_gradient(thermal->shapeDisplacement().space(), "shape_gradient");
+  shape_gradient = 0.0;
+  FiniteElementDual adjoint_load(thermal->temperature().space(), "adjoint_load");
+
+  for (int i = ts_info.num_timesteps; i > 0; --i) {
+    double             dt                      = ts_info.totalTime / ts_info.num_timesteps;
+    FiniteElementState temperature_end_of_step = thermal->previousTemperature(thermal->cycle());
+    computeStepAdjointLoad(temperature_end_of_step, adjoint_load, dt);
+    thermal->reverseAdjointTimestep({{"temperature", adjoint_load}});
+    shape_gradient += thermal->computeTimestepShapeSensitivity();
+  }
+
+  EXPECT_EQ(0, thermal->cycle());  // we are back to the start
+  auto initialConditionSensitivities     = thermal->computeInitialConditionSensitivity();
+  auto initialTemperatureSensitivityIter = initialConditionSensitivities.find("temperature");
+  SLIC_ASSERT_MSG(initialTemperatureSensitivityIter != initialConditionSensitivities.end(),
+                  "Could not find temperature in the computed initial condition sensitivities.");
+  initial_temperature_gradient += initialTemperatureSensitivityIter->second;
+
+  return std::make_tuple(qoi, initial_temperature_gradient, shape_gradient);
+}
+
+struct HeatTransferSensitivityFixture : public ::testing::Test {
+  void SetUp() override
+  {
+    MPI_Barrier(MPI_COMM_WORLD);
+    StateManager::initialize(dataStore, "thermal_dynamic_solve");
+    std::string filename = std::string(SERAC_REPO_DIR) + "/data/meshes/star.mesh";
+    mesh                 = StateManager::setMesh(mesh::refineAndDistribute(buildMeshFromFile(filename), 0));
+  }
+
+  void fillDirection(FiniteElementState& direction) const { direction = 1.1; }
+
+  // Create DataStore
+  axom::sidre::DataStore dataStore;
+  mfem::ParMesh*         mesh;
+
+  // Solver options
+  NonlinearSolverOptions nonlinear_opts{.relative_tol = 5.0e-13, .absolute_tol = 5.0e-13};
+  TimesteppingOptions    dyn_opts{.timestepper        = TimestepMethod::BackwardEuler,
+                               .enforcement_method = DirichletEnforcementMethod::DirectControl};
+  heat_transfer::IsotropicConductorWithLinearConductivityVsTemperature mat{1.0, 1.0, 1.0, 2.0};
+  TimeSteppingInfo                                                     tsInfo{.totalTime = 0.5, .num_timesteps = 4};
+};
+
+TEST_F(HeatTransferSensitivityFixture, InitialTemperatureSensitivities)
```

Comment:
Is a static parameterization (e.g. sensitivity with respect to a parameterized conductivity) test just not implemented or is there something fundamentally broken with that workflow?

Reply:
parametric sensitivies can be added for the next PR.  There is some more work to do to make those work correctly